cpp/src/io/orc/writer_impl.cu

```diff
@@ -1190,13 +1207,14 @@ orc_table_view make_orc_table_view(table_view const& table,
 
   rmm::device_uvector<orc_column_device_view> d_orc_columns(orc_columns.size(), stream);
 
+  rmm::device_uvector<thrust::pair<column_device_view const*, thrust::optional<uint32_t>>> stack(
+    50000, stream);
```

Comment:
This was just a test. I originally had this as a device local variable and kept increasing its size until it started erroring. Then I changed it to a single allocation in global mem.

We don't want table.num_columns(). That would break if we had a single struct with any children. We want the number of columns in this table and all its children and so on.

Reply:
Sorry, messed up the naming. meant `orc_columns.size()`, which includes all child columns as well.